src/pet.js

```diff
@@ -0,0 +1,69 @@
+const INITIAL_AGE = 0;
+const MINIMUM_HUNGER = 0;
+const MAX_FITNESS = 10;
+const MAX_AGE = 30;
+const MAX_HUNGER = 10;
+const MINIMUM_FITNESS = 0;
+
+function Pet(name) {
+  this.name = name;
+  this.age = INITIAL_AGE;
+  this.hunger = MINIMUM_HUNGER;
+  this.fitness = MAX_FITNESS;
+  this.children = [];
+}
+
+Pet.prototype = {
+  get isAlive() {
+    return (
+      this.age < MAX_AGE &&
+      this.hunger < MAX_HUNGER &&
+      this.fitness > MINIMUM_FITNESS
+    );
+  },
+};
+
+Pet.prototype.growUp = function () {
+  if (!this.isAlive) {
+    throw new Error("Your pet is no longer alive :(");
+  }
+  this.age += 1;
+  this.hunger += 5;
+  this.fitness -= 3;
+};
+
+Pet.prototype.walk = function () {
+  if (!this.isAlive) {
+    throw new Error("Your pet is no longer alive :(");
+  }
+  this.fitness = Math.min(this.fitness + 4, MAX_FITNESS);
```

Comment:
We could add more constants here for things like the fitness increment and threshold:

For example:
```
const AGE_INCREMENT = 1;
const FITNESS_INCREMENT = 4;
const HUNGER_INCREMENT = 3;
const HUNGER_THRESHOLD = 5
const FITNESS_THRESHOLD = 3
```
This function could then look like this:
```
  this.fitness = Math.min(this.fitness + FITNESS_INCREMENT, MAX_FITNESS);
```
This not only removes the hardcoded values from your functions, it also allows you to change these values in the future and only have to do it in your constants